packages/calcite-components/src/components/time-picker/time-picker.e2e.ts

```diff
@@ -1246,4 +1248,126 @@ describe("calcite-time-picker", () => {
+          expect(decimalSeparatorEl.textContent).toEqual(expectedLocalizedDecimalSeparator);
+          expect(fractionalSecondEl.textContent).toEqual(expectedLocalizedFractionalSecond);
+          if (secondSuffixEl) {
+            expect(secondSuffixEl.textContent).toEqual(expectedLocalizedSecondSuffix);
+          }
+          expect(meridiemEl.textContent).toEqual(expectedLocalizedMeridiem);
+        });
+        it("supports localized 24-hour format", async () => {
+          const initialDelocalizedValue = "14:02:30.001";
+          const page = await newE2EPage();
+          await page.setContent(html`
+            <calcite-time-picker
+              hour-format="24"
+              lang="${locale}"
+              step=".001"
+              value="${initialDelocalizedValue}"
+            ></calcite-time-picker>
+          `);
+
+          const {
+            localizedHour: expectedLocalizedHour,
+            localizedHourSuffix: expectedLocalizedHourSuffix,
+            localizedMinute: expectedLocalizedMinute,
+            localizedMinuteSuffix: expectedLocalizedMinuteSuffix,
+            localizedSecond: expectedLocalizedSecond,
+            localizedSecondSuffix: expectedLocalizedSecondSuffix,
+            localizedDecimalSeparator: expectedLocalizedDecimalSeparator,
+            localizedFractionalSecond: expectedLocalizedFractionalSecond,
+          } = localizeTimeStringToParts({
+            hour12: false,
+            value: initialDelocalizedValue,
+            locale,
+          });
+
+          const hourEl = await page.find(`calcite-time-picker >>> .${CSS.hour}`);
+          const hourSuffixEl = await page.find(`calcite-time-picker >>> .${CSS.hourSuffix}`);
+          const minuteEl = await page.find(`calcite-time-picker >>> .${CSS.minute}`);
+          const minuteSuffixEl = await page.find(`calcite-time-picker >>> .${CSS.minuteSuffix}`);
+          const secondEl = await page.find(`calcite-time-picker >>> .${CSS.second}`);
+          const decimalSeparatorEl = await page.find(`calcite-time-picker >>> .${CSS.decimalSeparator}`);
+          const fractionalSecondEl = await page.find(`calcite-time-picker >>> .${CSS.fractionalSecond}`);
+          const secondSuffixEl = await page.find(`calcite-time-picker >>> .${CSS.secondSuffix}`);
+          const meridiemEl = await page.find(`calcite-time-picker >>> .${CSS.meridiem}`);
+
+          expect(hourEl.textContent).toEqual(expectedLocalizedHour);
+          expect(hourSuffixEl.textContent).toEqual(expectedLocalizedHourSuffix);
+          expect(minuteEl.textContent).toEqual(expectedLocalizedMinute);
+          expect(minuteSuffixEl.textContent).toEqual(expectedLocalizedMinuteSuffix);
+          expect(secondEl.textContent).toEqual(expectedLocalizedSecond);
+          expect(decimalSeparatorEl.textContent).toEqual(expectedLocalizedDecimalSeparator);
+          expect(fractionalSecondEl.textContent).toEqual(expectedLocalizedFractionalSecond);
+          if (secondSuffixEl) {
```

Comment:
Bulgarian is the only one that I'm aware of (other than French Canada, which we don't officially support) that has a known suffix after the seconds, and I was told to strip that character out since we only are supporting short time formats for the time being.  I think its ok to keep this conditional in there, even if to remember to think about it for future locales being added that it might apply to.

Reply:
Sounds good. We can lean into the new vitest rule configs.

A comment with the above info on the conditional would be great!